Testsuite fails on big-endian systems

The testsuite fails during `test_focus_on_click` test with the following backtrace:

```
============================= test session starts ==============================
platform linux -- Python 3.9.7, pytest-6.2.5, py-1.10.0, pluggy-0.13.0
rootdir: /<<PKGBUILDDIR>>, configfile: tox.ini
plugins: forked-1.3.0, xvfb-1.2.0, xdist-2.4.0
collected 2129 items

../tests/test_basic_events.py E

==================================== ERRORS ====================================
_______________ ERROR at setup of test_focus_on_click[True-True] _______________

hlwm_process = <conftest.HlwmProcess object at 0x7fffb4c23040>
hlwm = <conftest.HlwmBridge object at 0x7fffb4c23df0>

    @pytest.fixture()
    def mouse(hlwm_process, hlwm):
        class Mouse:
            def __init__(self):
                self.move_to(0, 0, wait=True)
    
            def move_into(self, win_id, x=1, y=1, wait=True):
                if wait:
                    with hlwm_process.wait_stderr_match('EnterNotify'):
                        # no --sync here, because we're waiting for the EnterNotify anyways
                        self.call_cmd(f'xdotool mousemove --window {win_id} {x} {y}', shell=True)
                    # reaching this line only means that hlwm started processing
                    # the EnterNotify. So we need to wait until the event is fully processed:
                    hlwm.call('true')
                else:
                    self.call_cmd(f'xdotool mousemove --sync --window {win_id} {x} {y}', shell=True)
    
            def click(self, button, into_win_id=None, wait=True):
                if into_win_id:
                    # no need to wait for herbstluftwm to process, we're just positioning the mouse to click
                    self.move_into(into_win_id, wait=False)
                if wait:
                    with hlwm_process.wait_stderr_match('ButtonPress'):
                        subprocess.check_call(['xdotool', 'click', button])
                    # reaching this line only means that hlwm started processing
                    # the ButtonPress. So we need to wait until the event is fully processed:
                    hlwm.call('true')
                else:
                    subprocess.check_call(['xdotool', 'click', button])
    
            def move_to(self, abs_x, abs_y, wait=True):
                abs_x = str(int(abs_x))
                abs_y = str(int(abs_y))
                self.call_cmd(f'xdotool mousemove --sync {abs_x} {abs_y}', shell=True)
                if wait:
                    # wait until all the mouse move events that are now in the queue
                    # are fully processed:
                    hlwm.call('true')
    
            def move_relative(self, delta_x, delta_y, wait=True):
                self.call_cmd(f'xdotool mousemove_relative --sync {delta_x} {delta_y}', shell=True)
                if wait:
                    # wait until all the mouse move events that were put in the
                    # queue by the above xdotool invokation are fully processed.
                    # (other than for the button events, the motion notify events
                    # are not printed to stderr, because this would lead to
                    # to much debug output on motions of the physical mouse)
                    hlwm.call('true')
    
            def mouse_press(self, button, wait=True):
                cmd = ['xdotool', 'mousedown', button]
                if wait:
                    with hlwm_process.wait_stderr_match('ButtonPress'):
                        subprocess.check_call(cmd)
                    # wait for the ButtonPress to be fully processed:
                    hlwm.call('true')
                else:
                    subprocess.check_call(cmd)
    
            def mouse_release(self, button, wait=True):
                cmd = ['xdotool', 'mouseup', button]
                if wait:
                    with hlwm_process.wait_stderr_match('ButtonRelease'):
                        subprocess.check_call(cmd)
                    # wait for the ButtonRelease to be processed
                    hlwm.call('true')
                else:
                    subprocess.check_call(cmd)
    
            def call_cmd(self, cmd, shell=False):
                print('calling: {}'.format(cmd), file=sys.stderr)
                subprocess.check_call(cmd, shell=shell)
    
>       return Mouse()

../tests/conftest.py:1081: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../tests/conftest.py:1012: in __init__
    self.move_to(0, 0, wait=True)
../tests/conftest.py:1045: in move_to
    hlwm.call('true')
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <conftest.HlwmBridge object at 0x7fffb4c23df0>, cmd = 'true'
allowed_stderr = None

    def call(self, cmd, allowed_stderr=None):
        """call the command and expect it to have exit code zero
        and no output on stderr.
        if allowed_stderr is set, then it should be a regex object
        (something with .match) that matches every line the command
        writes to stderr.
        """
        proc = self.unchecked_call(cmd)
>       assert proc.returncode == 0
E       AssertionError
```

Full log available at: https://buildd.debian.org/status/fetch.php?pkg=herbstluftwm&arch=ppc64&ver=0.9.3-2&stamp=1635420861&raw=0

Access to big-endian machines for developers interested in fixing this issue can be obtained through the gcc compile farm, see: https://gcc.gnu.org/wiki/CompileFarm

--- a/NEWS
+++ b/NEWS
@@ -16,7 +16,7 @@ Current git version
     are shown. This should be used instead of the old 'always_show_frame'
   * New frame attribute 'content_geometry'
   * New monitor attribute 'content_geometry'
-  * Fix bug in ipc protocoll for big-endian systems
+  * Fix bug in ipc protocol for big-endian systems
 
 Release 0.9.4 on 2022-03-16
 ---------------------------

--- a/NEWS
+++ b/NEWS
@@ -16,6 +16,7 @@ Current git version
     are shown. This should be used instead of the old 'always_show_frame'
   * New frame attribute 'content_geometry'
   * New monitor attribute 'content_geometry'
+  * Fix bug in ipc protocoll for big-endian systems
 
 Release 0.9.4 on 2022-03-16
 ---------------------------

--- a/src/ipc-server.cpp
+++ b/src/ipc-server.cpp
@@ -52,7 +52,9 @@ bool IpcServer::handleConnection(Window win, CallHandler callback) {
     }
     auto result = callback(maybeArguments.value());
     // send output back
-    int status = result.exitCode;
+    // here, it is important to use 'long' because XChangeProperty() requires
+    // data of type long.
+    long status = static_cast<long>(result.exitCode);
     // Mark this command as executed
     XDeleteProperty(X.display(), win, X.atom(HERBST_IPC_ARGS_ATOM));
     X.setPropertyString(win, X.atom(HERBST_IPC_OUTPUT_ATOM), result.output);

